### PR TITLE
Changeset for getRelaySessionInfo and providers for it

### PR DESCRIPTION
**Pull request by ghoshkaj**
Add changeset for https://github.com/microsoft/FluidFramework/pull/16242, https://github.com/microsoft/FluidFramework/pull/16160

### CODE REVIEW

File: .changeset/chubby-schools-jump.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+"@fluidframework/odsp-driver": minor
+"@fluidframework/odsp-driver-definitions": minor
+---
+
+A new API getRelaySessioInfo to surface relay session info from cache, and providers to allow access to the new API
```
Comment (connorskees) [inline .changeset/chubby-schools-jump.md:6]: Did you mean to include the `` character? Same question in other file

Comment (ghoshkaj) [reply .changeset/chubby-schools-jump.md:6]: Odd, I don't see the character you're referring to in the text.

Comment (connorskees) [reply .changeset/chubby-schools-jump.md:6]: Interesting -- 
image
Let me look at some other changeset files

Comment (ghoshkaj) [reply .changeset/chubby-schools-jump.md]: But I found it in visual studio. Got rid of it!

**Review by connorskees — APPROVED** [review]
seems sane to me, though i've no context on the specific feature

File: .changeset/chubby-schools-jump.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+"@fluidframework/odsp-driver": minor
+"@fluidframework/odsp-driver-definitions": minor
+---
+
+A new API getRelaySessionInfo to surface relay session info from cache, and providers to allow access to the new API
```
Comment (alexvy86) [inline .changeset/chubby-schools-jump.md:6]: Nit:
```suggestion
`OdspDocumentServiceFactory` and `OdspDocumentServiceFactoryCore` acquired a new API `getRelaySessionInfo` to surface relay session info from cache, and implement new interfaces `IRelaySessionAwareDriverFactory` and `IProvideSessionAwareDriverFactory` that enable using the provider pattern to discover the new API.
```

**Review by alexvy86 — APPROVED** [review]
Nit on the phrasing. Also, Is it expected that we have two identical lines in the changeset file?

**Comment by ghoshkaj** [comment]
> Also, Is it expected that we have two identical lines in the changeset file?

@alexvy86 it was autogenerated from the cli tool from one line input of mine. I suspect that the first line is for the changeset of odsp-driver and the second for odsp-driver-definitions. So I've separated them out accordingly.

**Comment by alexvy86** [comment]
>> Also, Is it expected that we have two identical lines in the changeset file?

> @alexvy86 it was autogenerated from the cli tool from one line input of mine. I suspect that the first line is for the changeset of odsp-driver and the second for odsp-driver-definitions. So I've separated them out accordingly.

Pinging @tylerbutler to get a better sense of that, and what should we do about it by hand, if anything. If it is one line/paragraph per touched package, I'm curious how multi-paragraph changesets for a package would work.

**Comment by tylerbutler** [comment]
> > > Also, Is it expected that we have two identical lines in the changeset file?
> 
> > @alexvy86 it was autogenerated from the cli tool from one line input of mine. I suspect that the first line is for the changeset of odsp-driver and the second for odsp-driver-definitions. So I've separated them out accordingly.
> 
> Pinging @tylerbutler to get a better sense of that, and what should we do about it by hand, if anything. If it is one line/paragraph per touched package, I'm curious how multi-paragraph changesets for a package would work.

You can edit the changeset files by hand, yes. But UPCOMING.md should always be generated using `pnpm run generate:upcoming`.

**Comment by tylerbutler** [comment]
The changeset as a whole applies to all the packages in the metadata. If you want to have different changelog entries per package, create two changesets, one for each package.

**Review by tylerbutler — APPROVED** [review]
@ghoshkaj I pushed an update to this PR. If the edits look good to you, go ahead and merge.

**Pull request merged**